ngraph/frontend/paddlepaddle/src/place.cpp

```diff
@@ -25,26 +25,226 @@ bool PlacePDPD::is_output() const
 }
 
 OpPlacePDPD::OpPlacePDPD(const InputModel& input_model,
-                         const std::vector<std::string>& names,
-                         const std::shared_ptr<paddle::framework::proto::OpDesc>& op_desc)
+                         const paddle::framework::proto::OpDesc& op_desc,
+                         const std::vector<std::string>& names)
     : PlacePDPD(input_model, names)
     , m_op_desc(op_desc)
 {
 }
 
 OpPlacePDPD::OpPlacePDPD(const InputModel& input_model,
-                         const std::shared_ptr<paddle::framework::proto::OpDesc>& op_desc)
-    : OpPlacePDPD(input_model, {}, op_desc)
+                         const paddle::framework::proto::OpDesc& op_desc)
+    : OpPlacePDPD(input_model, op_desc, {})
 {
 }
 
+const std::map<std::string, std::vector<std::shared_ptr<OutPortPlacePDPD>>>&
+    OpPlacePDPD::get_output_ports() const
+{
+    return m_output_ports;
+}
+
+const std::map<std::string, std::vector<std::shared_ptr<InPortPlacePDPD>>>&
+    OpPlacePDPD::get_input_ports() const
+{
+    return m_input_ports;
+}
+
+std::shared_ptr<OutPortPlacePDPD> OpPlacePDPD::get_output_port_pdpd(const std::string& name,
+                                                                    int idx) const
+{
+    return m_output_ports.at(name)[idx];
```

Comment:
Should there be some FRONT_END_GENERAL_CHECK to ensure that we're in bounds?

Reply:
ok, it can be useful in some cases

Reply:
done